<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingsDialogForm</class>
 <widget class="QDialog" name="SettingsDialogForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>836</width>
    <height>778</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>600</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../galacteek.qrc">
    <normaloff>:/share/icons/settings.png</normaloff>:/share/icons/settings.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>3</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>General</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="1" column="1" colspan="2">
        <widget class="QLineEdit" name="home">
         <property name="minimumSize">
          <size>
           <width>400</width>
           <height>0</height>
          </size>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Home:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="6" column="1" colspan="2">
        <spacer>
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>391</width>
           <height>262</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="5" column="1">
        <widget class="QPushButton" name="changeDownloadsPathButton">
         <property name="maximumSize">
          <size>
           <width>200</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="text">
          <string>Change directory</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Save downloads to:</string>
         </property>
         <property name="alignment">
          <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
         </property>
        </widget>
       </item>
       <item row="4" column="1" colspan="2">
        <widget class="QLineEdit" name="downloadsLocation"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_6">
      <attribute name="icon">
       <iconset resource="../galacteek.qrc">
        <normaloff>:/share/icons/stroke-cube.png</normaloff>:/share/icons/stroke-cube.png</iconset>
      </attribute>
      <attribute name="title">
       <string>IPFS</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_5">
         <property name="sizeConstraint">
          <enum>QLayout::SetNoConstraint</enum>
         </property>
         <item>
          <widget class="QGroupBox" name="groupBoxDaemon">
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>50</height>
            </size>
           </property>
           <property name="title">
            <string>Spawn IPFS daemon and use as transport</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <layout class="QGridLayout" name="gridLayout_5">
            <item row="1" column="3">
             <widget class="QSpinBox" name="ipfsdGwPort">
              <property name="maximumSize">
               <size>
                <width>100</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="minimum">
               <number>1024</number>
              </property>
              <property name="maximum">
               <number>32768</number>
              </property>
             </widget>
            </item>
            <item row="8" column="1">
             <widget class="QComboBox" name="pubsubRoutingMode">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Pusub routing mode (gossipsub is more bandwidth-efficient)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_20">
              <property name="text">
               <string>Swarm port (QUIC)</string>
              </property>
             </widget>
            </item>
            <item row="4" column="1">
             <widget class="QSpinBox" name="swarmMinConns">
              <property name="minimum">
               <number>20</number>
              </property>
              <property name="maximum">
               <number>1000</number>
              </property>
              <property name="singleStep">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item row="1" column="2">
             <widget class="QLabel" name="label_17">
              <property name="text">
               <string>HTTP gateway port</string>
              </property>
             </widget>
            </item>
            <item row="8" column="0">
             <widget class="QLabel" name="label_13">
              <property name="text">
               <string>Pubsub routing</string>
              </property>
             </widget>
            </item>
            <item row="7" column="1">
             <widget class="QComboBox" name="routingMode">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Routing mode (dht is the default, dhtclient does not act as a full node on the DHT)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <item>
               <property name="text">
                <string>dht</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>dhtclient</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="ipfsdSwarmPortQuic">
              <property name="minimum">
               <number>1024</number>
              </property>
              <property name="maximum">
               <number>32768</number>
              </property>
             </widget>
            </item>
            <item row="5" column="2">
             <widget class="QCheckBox" name="writableHttpGw">
              <property name="text">
               <string>Writable HTTP gateway</string>
              </property>
             </widget>
            </item>
            <item row="3" column="0" colspan="4">
             <widget class="Line" name="line">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
             </widget>
            </item>
            <item row="4" column="0">
             <widget class="QLabel" name="label_18">
              <property name="text">
               <string>Minimum swarm connections</string>
              </property>
             </widget>
            </item>
            <item row="0" column="3">
             <widget class="QSpinBox" name="ipfsdApiPort">
              <property name="maximumSize">
               <size>
                <width>100</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="minimum">
               <number>1024</number>
              </property>
              <property name="maximum">
               <number>32768</number>
              </property>
             </widget>
            </item>
            <item row="7" column="0">
             <widget class="QLabel" name="label_8">
              <property name="text">
               <string>Routing mode</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QSpinBox" name="ipfsdSwarmPort">
              <property name="maximumSize">
               <size>
                <width>100</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="minimum">
               <number>1024</number>
              </property>
              <property name="maximum">
               <number>32768</number>
              </property>
             </widget>
            </item>
            <item row="8" column="2">
             <widget class="QCheckBox" name="fileStore">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Use the IPFS Filestore system (avoids file duplication)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="text">
               <string>Enable Filestore</string>
              </property>
             </widget>
            </item>
            <item row="5" column="1">
             <widget class="QSpinBox" name="swarmMaxConns">
              <property name="minimum">
               <number>150</number>
              </property>
              <property name="maximum">
               <number>2000</number>
              </property>
              <property name="singleStep">
               <number>5</number>
              </property>
             </widget>
            </item>
            <item row="4" column="2">
             <widget class="QLabel" name="label_5">
              <property name="text">
               <string>Maximum storage (Gb)</string>
              </property>
             </widget>
            </item>
            <item row="4" column="3">
             <widget class="QSpinBox" name="storageMax">
              <property name="minimum">
               <number>5</number>
              </property>
              <property name="maximum">
               <number>2048</number>
              </property>
             </widget>
            </item>
            <item row="5" column="0">
             <widget class="QLabel" name="label_19">
              <property name="text">
               <string>Maximum swarm connections</string>
              </property>
             </widget>
            </item>
            <item row="0" column="0">
             <widget class="QLabel" name="label_14">
              <property name="text">
               <string>Swarm port</string>
              </property>
             </widget>
            </item>
            <item row="9" column="2">
             <widget class="QLabel" name="label_9">
              <property name="text">
               <string/>
              </property>
             </widget>
            </item>
            <item row="0" column="2">
             <widget class="QLabel" name="label_15">
              <property name="text">
               <string>API port</string>
              </property>
             </widget>
            </item>
            <item row="7" column="2">
             <widget class="QCheckBox" name="namesysPubsub">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Enable IPNS over pubsub (updating of IPNS records over pubsub channels)&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="text">
               <string>Enable IPNS over Pubsub</string>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QCheckBox" name="checkBoxQuic">
              <property name="text">
               <string>Swarm: enable QUIC</string>
              </property>
             </widget>
            </item>
            <item row="9" column="0">
             <widget class="QCheckBox" name="keepDaemonRunning">
              <property name="toolTip">
               <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Always keep the IPFS daemon running.&lt;/p&gt;&lt;p&gt;This ensures that your files are still accessible even when galacteek is not running.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
              </property>
              <property name="text">
               <string>Keep IPFS daemon running</string>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
         <item>
          <widget class="QGroupBox" name="groupBoxIpfsConn">
           <property name="enabled">
            <bool>true</bool>
           </property>
           <property name="minimumSize">
            <size>
             <width>0</width>
             <height>50</height>
            </size>
           </property>
           <property name="title">
            <string>Use a custom IPFS daemon</string>
           </property>
           <property name="flat">
            <bool>false</bool>
           </property>
           <property name="checkable">
            <bool>false</bool>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
           <layout class="QFormLayout" name="formLayout_3">
            <item row="0" column="0">
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Host</string>
              </property>
             </widget>
            </item>
            <item row="0" column="1">
             <widget class="QLineEdit" name="customIpfsHost">
              <property name="maximumSize">
               <size>
                <width>300</width>
                <height>16777215</height>
               </size>
              </property>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_16">
              <property name="text">
               <string>API port</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QSpinBox" name="customIpfsApiPort">
              <property name="maximumSize">
               <size>
                <width>100</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="minimum">
               <number>1024</number>
              </property>
              <property name="maximum">
               <number>32768</number>
              </property>
             </widget>
            </item>
            <item row="2" column="0">
             <widget class="QLabel" name="label">
              <property name="text">
               <string>HTTP gateway port</string>
              </property>
             </widget>
            </item>
            <item row="2" column="1">
             <widget class="QSpinBox" name="customIpfsGwPort">
              <property name="maximumSize">
               <size>
                <width>100</width>
                <height>16777215</height>
               </size>
              </property>
              <property name="minimum">
               <number>1024</number>
              </property>
              <property name="maximum">
               <number>32768</number>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_4">
           <property name="maximumSize">
            <size>
             <width>721</width>
             <height>145</height>
            </size>
           </property>
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>Restart the application after changing the IPFS connection settings.</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="icon">
       <iconset resource="../galacteek.qrc">
        <normaloff>:/share/icons/ethereum.png</normaloff>:/share/icons/ethereum.png</iconset>
      </attribute>
      <attribute name="title">
       <string>Ethereum</string>
      </attribute>
      <layout class="QVBoxLayout" name="verticalLayout_3">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="QGroupBox" name="groupBoxEth">
           <property name="title">
            <string>Enable Ethereum connector</string>
           </property>
           <property name="checkable">
            <bool>true</bool>
           </property>
           <property name="checked">
            <bool>false</bool>
           </property>
           <layout class="QGridLayout" name="gridLayout_8">
            <item row="0" column="0">
             <layout class="QGridLayout" name="gridLayout_7">
              <item row="0" column="0">
               <widget class="QLabel" name="label_10">
                <property name="text">
                 <string>Provider type</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item row="0" column="1" alignment="Qt::AlignRight">
             <widget class="QComboBox" name="ethProvType">
              <property name="maximumSize">
               <size>
                <width>500</width>
                <height>16777215</height>
               </size>
              </property>
              <item>
               <property name="text">
                <string>http</string>
               </property>
              </item>
              <item>
               <property name="text">
                <string>websocket</string>
               </property>
              </item>
             </widget>
            </item>
            <item row="1" column="0">
             <widget class="QLabel" name="label_11">
              <property name="text">
               <string>RPC URL</string>
              </property>
             </widget>
            </item>
            <item row="1" column="1">
             <widget class="QLineEdit" name="ethRpcUrl">
              <property name="maximumSize">
               <size>
                <width>500</width>
                <height>16777215</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <spacer name="verticalSpacer">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="icon">
       <iconset resource="../galacteek.qrc">
        <normaloff>:/share/icons/window.png</normaloff>:/share/icons/window.png</iconset>
      </attribute>
      <attribute name="title">
       <string>UI</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <spacer name="verticalSpacer_2">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>40</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="../galacteek.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>SettingsDialogForm</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>248</x>
     <y>254</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>SettingsDialogForm</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>316</x>
     <y>260</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
